### PR TITLE
[14.0][MIG] project

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
Migration done.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
I have pushed changes to finish/fix the scripts.

File: openupgrade_scripts/scripts/project/14.0.1.1/pre-migration.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        env.cr,
+        [
+            ("project.access_partner_task user", "project.access_partner_task_user"),
+        ],
+    )
+    _set_default_rating_status_period(env)
+    # Add temporary tables for avoiding the automatic launch of the compute or default
+    openupgrade.logged_query(
+        env.cr,
+        "CREATE TABLE project_allowed_internal_users_rel (temp INTEGER)",
```
Comment (MiquelRForgeFlow) [inline openupgrade_scripts/scripts/project/14.0.1.1/pre-migration.py]: CI says you should add the columns

Comment (pedrobaeza) [reply openupgrade_scripts/scripts/project/14.0.1.1/pre-migration.py]: It seems Odoo has changed now the way it reflects the FK and indexes, so we only need to create the table at pre.

File: openupgrade_scripts/scripts/project/14.0.1.1/pre-migration.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    # Manually create tables for avoiding the automatic launch of the compute or default
+    # FK constraints and indexes will be added by ORM
+    openupgrade.logged_query(
+        env.cr,
+        """CREATE TABLE project_allowed_internal_users_rel
+        (project_project_id INTEGER, res_users_id INTEGER)""",
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """CREATE TABLE project_task_res_users_rel
```
Comment (MiquelRForgeFlow) [inline openupgrade_scripts/scripts/project/14.0.1.1/pre-migration.py:52]:
```suggestion
        """CREATE TABLE project_allowed_portal_users_rel
```

Comment (pedrobaeza) [reply openupgrade_scripts/scripts/project/14.0.1.1/pre-migration.py:52]: No, that table doesn't have default, but the other has a computed.

File: openupgrade_scripts/scripts/project/14.0.1.1/pre-migration.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        (project_project_id INTEGER, res_users_id INTEGER)""",
```
Comment (MiquelRForgeFlow) [inline openupgrade_scripts/scripts/project/14.0.1.1/pre-migration.py]:
```suggestion
        (project_task_id INTEGER, res_users_id INTEGER)""",
```

File: openupgrade_scripts/scripts/project/14.0.1.1/post-migration.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    # TODO: Take into account channels and task followers part of the old rule
+    openupgrade.logged_query(
+        env.cr,
+        """
+        INSERT INTO project_allowed_internal_users_rel
+        (project_project_id, res_users_id)
+        SELECT pp.id, ru.id
+        FROM project_project pp
+        JOIN mail_followers mf ON mf.res_model = 'project.project' AND mf.res_id = pp.id
+        JOIN res_users ru ON ru.partner_id = rmf.partner_id AND NOT ru.share
```
Comment (MiquelRForgeFlow) [inline openupgrade_scripts/scripts/project/14.0.1.1/post-migration.py]:
```suggestion
        JOIN res_users ru ON ru.partner_id = mf.partner_id AND NOT ru.share
```

File: openupgrade_scripts/scripts/project/14.0.1.1/post-migration.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+        """
+        INSERT INTO project_task_res_users_rel
+        (project_task_id, res_users_id)
+        SELECT pt.id, rel.res_users_id
+        FROM project_allowed_internal_users_rel rel
+        JOIN project_task pt WHERE pt.project_id = rel.project_project_id
+        UNION
+        SELECT pt.id, rel.res_users_id
+        FROM project_allowed_portal_users_rel rel
+        JOIN project_task pt WHERE pt.project_id = rel.project_project_id
```
Comment (MiquelRForgeFlow) [inline openupgrade_scripts/scripts/project/14.0.1.1/post-migration.py]:
```suggestion
        JOIN project_task pt ON pt.project_id = rel.project_project_id
        UNION
        SELECT pt.id, rel.res_users_id
        FROM project_allowed_portal_users_rel rel
        JOIN project_task pt ON pt.project_id = rel.project_project_id
```

Comment (pedrobaeza) [reply openupgrade_scripts/scripts/project/14.0.1.1/post-migration.py]: As I'm doing this between training sessions, I haven't put enough effort on checking SQLs. Sorry for the hammering.

**Comment by pedrobaeza** [comment]
Is it OK for you now, @MiquelRForgeFlow ?

**Comment by MiquelRForgeFlow** [comment]
I suppose, yes :+1:

**Pull request merged**